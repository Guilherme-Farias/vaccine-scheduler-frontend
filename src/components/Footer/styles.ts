import styled, { css } from 'styled-components/macro';

export const Container = styled.footer`
  ${({ theme }) => css`
    padding: ${theme.spacings.lg} 0;

    @media ${theme.queries.phoneAndSmaller} {
      padding: ${theme.spacings.xs} 0;
    }
  `}
`;

export const Copyright = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.black};
    font-size: ${theme.font.sizes.lg};
    margin: auto;
    text-align: center;

    @media ${theme.queries.phoneAndSmaller} {
      font-size: ${theme.font.sizes.md};
    }
  `}
`;
