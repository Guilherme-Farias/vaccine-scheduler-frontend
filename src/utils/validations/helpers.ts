import Joi from 'joi';
import { IParamErrors } from 'types/global';

export function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: IParamErrors = {};

  if (objError.error) {
    objError.error.details.forEach(err => {
      errors[err.path.join('.')] = err.message;
    });
  }

  return errors;
}
