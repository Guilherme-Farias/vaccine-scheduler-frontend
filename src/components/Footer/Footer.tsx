import React from 'react';

import MaxWidthWrapper from 'components/MaxWidthWrapper';
import * as S from './styles';

const currentYear = new Date().getFullYear();

const Footer: React.FC = () => (
  <S.Container>
    <MaxWidthWrapper>
      <S.Copyright>
        Â© {currentYear} Minha Vacina. Todos os direitos reservados.
      </S.Copyright>
    </MaxWidthWrapper>
  </S.Container>
);

export default Footer;
