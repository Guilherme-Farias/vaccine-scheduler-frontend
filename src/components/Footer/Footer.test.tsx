import { render, screen } from 'utils/tests/helpers';

import Footer from '.';

describe('<Footer />', () => {
  it('should render the Copyright', () => {
    const { container } = render(<Footer />);

    const copyrightRegex =
      /^Â© (\d{4}) Minha Vacina\. Todos os direitos reservados\.$/i;

    expect(screen.getByText(copyrightRegex)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
