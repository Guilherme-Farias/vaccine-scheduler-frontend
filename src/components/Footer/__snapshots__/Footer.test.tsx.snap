// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the Copyright 1`] = `
.c1 {
  width: 100%;
  max-width: 81.25rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
}

.c0 {
  padding: 1.25rem 0;
}

.c2 {
  color: #171923;
  font-size: 1.125rem;
  margin: auto;
  text-align: center;
}

@media (max-width:48rem) {
  .c0 {
    padding: 0.5rem 0;
  }
}

@media (max-width:48rem) {
  .c2 {
    font-size: 1rem;
  }
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      Â© 
      2022
       Minha Vacina. Todos os direitos reservados.
    </p>
  </div>
</footer>
`;
